### PR TITLE
chore: Introduce analytics in Extensible CLI

### PR DESCRIPTION
**Pull request by PeterSchafer**
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/CONTRIBUTING.md) rules

#### What does this PR do?
This PR adds a basic implementation of the analytics module in golang. Therefore introduces a global configuration package to access configuration from different sources in a unified way. 
And some minor refactorings to reduce coupling and avoid cyclic dependencies.

#### Where should the reviewer start?
For the integration of the feature the best location is `cliv2/cmd/cliv2/main.go`.

#### How should this be manually tested?
Run any command and use for example logz.io to see the events arrive. Commands that execute in the legacy CLI will create two analytic entries.

#### What are the relevant tickets?
HMMR-528

### CODE REVIEW

File: cliv2/cmd/cliv2/main.go [hunk 1]
```diff
@@ -75,39 +115,47 @@ func MainWithErrorCode(envVariables EnvironmentVariables, args []string) int {
...
 	if err != nil {
 		fmt.Println("Failed to create proxy")
 		fmt.Println(err)
-		return cliv2.SNYK_EXIT_CODE_ERROR
+		cliAnalytics.AddError(err)
+		return constants.SNYK_EXIT_CODE_ERROR
 	}
 
 	wrapperProxy.SetUpstreamProxyAuthentication(envVariables.ProxyAuthenticationMechanism)
+	http.DefaultTransport = wrapperProxy.Transport()
```
Comment (Avishagp) [inline cliv2/cmd/cliv2/main.go:141]: Why was this added?

Comment (PeterSchafer) [reply cliv2/cmd/cliv2/main.go:141]: This ensures that the global DefaultTransport uses the Kerberos Proxy Authentication. Since Analytics traffic in this case doesn't go through the internal proxy. Which would technically work but would add unreasonable extra computations.
